flow/include/flow/CoroutinesImpl.h

```diff
@@ -270,6 +271,74 @@ struct AwaitableFuture : std::conditional_t<IsStream, SingleCallback<ToFutureVal
 	}
 };
 
+// TODO: This can be merged with AwaitableFutureStream by passing more template arguments.
```

Comment:
Why not do it?

Reply:
Soon, once I test the notified queue, promise, future. I wanted to see if there cleaner way than nested `std::conditional_t` to select appropriate type in `AwaitableFutureStream` as we will have more than one conditions, which is kind of little diversion.

Reply:
ack!